bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/NMSSM_XYH_ggZZ/Template/proc_card.dat

```diff
@@ -0,0 +1,22 @@
+set group_subprocesses Auto
+set ignore_six_quark_processes False
+set loop_optimized_output True
+set loop_color_flows False
+set gauge unitary
+set complex_mass_scheme False
+set max_npoint_for_channel 0
+import model sm
+define p = g u c d s u~ c~ d~ s~
+define j = g u c d s u~ c~ d~ s~
+define l+ = e+ mu+
+define l- = e- mu-
+define vl = ve vm vt
+define vl~ = ve~ vm~ vt~
+set group_subprocesses Auto
+set ignore_six_quark_processes False
+set loop_optimized_output True
+set gauge unitary
+set complex_mass_scheme False
+import model NMSSMHET_UFO --modelname
+generate g g > h03 , (h03 > h02 h01, h02 > tau+ tau- all all, h01 > a a)
```

Comment:
"h02 > tau+ tau- all all" is so vague syntax
Can you let us know what are the final states you target for?

Reply:
The final state we are interested in tau+ tau- gamma gamma. The reasoning behind the ggZZ cards is that suppose you have a process where H->gg and H->ZZ, then if one of the Z bosons decay into tau+ tau-, then you have your tau+ tau- gamma gamma final state + some 2 extra particles from the other Z boson. 

At least this was my initial intention/idea. I thought that I ought not to restrict myself to the H->ZZ decay in particular. For example, you could also get H->WW-> tau+ nu tau- nu which would lead to the same final state. By using "h02 > tau+ tau- all all", I am looking for tau+ tau- gamma gamma + some 2 extra particles from anywhere.

Reply:
But you get bunch of contributions from diagrams that you might not want and increase the computing time so much.
If "m(h02) > 2 X m(Z or W)" would you not instead try to specify the decay channels you want?
By doing something like (define sth = leptons and quarks)
h02 > z z, (z > tau+ tau-), (z > sth sth)
h02 > w+ w-, (w+ > tau+ ......

<img width="508" alt="image" src="https://user-images.githubusercontent.com/17442656/126339058-0a18ce32-a432-4afe-80bf-c09de8bea1ee.png">

Reply:
I originally tried these different process lines in the sm model and saw only approx. 200 diagrams for "h > tau+ tau- all all" which is why I thought it might not be an issue. I've since tried it in the NMSSMHET_UFO and now I get closer to 10,000 so I agree that this is too vague. 

I am now unsure what process line to write. I was confused about the suggestion of "h02 > z z, (z > tau+ tau-), (z > sth sth)" since my understanding is that this process line would require the z bosons to be on-shell. I do not think this requirement is necessary. An alternative would be "h02 > z z > ta+ ta- sth sth" but then I have also been told that this sort of syntax can lead to gauge invariance issues. What would you suggest to do?

Reply:
Since I guess you'll be triggering on taus which would require some pT cuts, I don't think requiring onshell Z will give you too much loss (pTtau >~ 30 GeV). But then it would be safer to keep the two sths to come from offshell Z/gamma.
What I would try is
"h02 > z sth sth, z > ta+ ta-" where "sth = lepton, neutrino, and jet"
"h02 > w+ w-, w+ > ta+ vt, w- > ta- vt~"

Does this make sense to you?
If your are concerned, you might want to do a quick check with the cross section differences between a. and b. and see how much offshell contributions will be.
a. h02 > e+ e- ta+ ta-
b. h02 > z z, z > e+ e-, z > ta+ ta-

Reply:
We want to derive samples at multiple values of the h02 (Y) mass (see generate_grid.py L74). Some of these values are below 2*m_Z, in which case it won't be possible for both of the Z bosons to be on-shell. I tried calculating the partial widths you suggested anyway. 

For h02 mass = 90 GeV, and h03 mass = 300 GeV I got:
a. 0.425 pm 0.006 GeV
b. 7.38e-15 pm 2.43e-17 GeV
So assuming I haven't made a mistake, this shows the off-shell contributions are neccessary?

Reply:
Sorry for the late response.

I am not sure what you did. If you set mh02 = 90, then
"h02 > z z, z > e+ e-, z > ta+ ta-" does not make sense at all, did you do this? If so, I am actually a bit surprised to see this worked. Because as you said, b. is only possible for MG to calculate when 2*mZ < mh02.

I can't think of any smart tricks to do mh02 = 90 GeV production yet.
What would be sensible to do for the mass points above will be checking the offshell contributions by comparing :
a. generate h02 > z e+ e-, z > ta+ ta-
b. generate h02 > ta+ ta- e+ e-
When MadGraph asks you to edit run card, you might have to type in "set no_parton_cut" and remove all kinematic cuts, and then apply minimal mmll or ptl cuts (~5 GeV maybe) as the width calculations will be affected by the default kinematic cut settings in MadGraph.

If you are not sure on what to do, please send me an email shjeon@cernREMOVETHIS.ch